middle_end/flambda2/cmx/flambda_cmx.ml

```diff
@@ -176,3 +241,26 @@ let prepare_cmx_file_contents ~final_typing_env ~module_symbol
     Some
       (Flambda_cmx_format.create ~final_typing_env ~all_code ~exported_offsets
          ~used_closure_vars)
+
+let prepare_cmx_from_approx ~approxs ~exported_offsets ~used_closure_vars
+    all_code =
+  if Flambda_features.opaque ()
```

Comment:
We could add a `|| not (Flambda_features.classic_mode ())` to save some time when not in classic mode.